pkg/server/server.go

```diff
@@ -90,6 +93,62 @@ func (s *Server) SetRules(w http.ResponseWriter, r *http.Request, tenant string)
 	_, _ = w.Write([]byte("successfully updated rules file"))
 }
 
+func (s *Server) ListAllRules(w http.ResponseWriter, r *http.Request) {
+	logger := log.With(s.logger, "handler", "listAllRules")
+
+	//nolint:exhaustivestruct
+	allGroups := &rulefmt.RuleGroups{}
+
+	_ = s.bucket.Iter(r.Context(), rulesBasePath, func(dir string) error {
+		tenant := strings.TrimPrefix(dir, rulesBasePath)
+		tenant = strings.TrimSuffix(tenant, "/")
+
+		file, err := s.bucket.Get(r.Context(), getRulesFilePath(tenant))
+		if err != nil {
+			level.Warn(logger).Log("msg", "failed retrieving rules file", "tenant", tenant, "err", err)
+
+			return nil
+		}
+		defer file.Close()
+
+		data, err := ioutil.ReadAll(file)
+		if err != nil {
+			level.Warn(logger).Log("msg", "error reading rules file", "tenant", tenant, "err", err)
+
+			return nil
```

Comment:
Hmm i worry that it's not prudent to ignore all of these errors and to not return a 500 to the caller. Failing to read these files is probably a serious problem: the files are on disk but the kernel is misbehaving.

Reply:
Got it. Now I am returning `500 Internal Server Error` if this happens. Thanks!